src/Control/Effect/Labelled.hs

```diff
@@ -0,0 +1,80 @@
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE FunctionalDependencies #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE PolyKinds #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilyDependencies #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE UndecidableInstances #-}
+module Control.Effect.Labelled
+( runLabelled
+, Labelled(Labelled)
+, LabelledMember(..)
+, HasLabelled
+, runUnderLabel
+, UnderLabel(UnderLabel)
+, module Control.Algebra
+) where
+
+import Control.Algebra
+import Control.Monad.IO.Class
+import Control.Monad.Trans.Class
+
+newtype Labelled (label :: k) (sub :: (* -> *) -> (* -> *)) m a = Labelled { runLabelled :: sub m a }
+  deriving (Applicative, Effect, Functor, HFunctor, Monad, MonadFail, MonadIO, MonadTrans)
```

Comment:
> Both a particle, and a wave.

`Labelled` serves both to wrap unlabelled effects, labelling them within a signature, and as a carrier which lifts carriers for unlabelled effects into carriers for (the same) labelled effects.

I’ve known for a while that monad transformers and effects have the same kind, and so e.g. it is technically possible to use `IdentityT`, `Ap`, or `Alt` as an effect. But to what end?

This, by contrast, is both an effect transformer and a monad transformer transformer (NB: not a typo), and derives all of its utility from the phantom `label` parameter, which the rest of this module associates with a single effect within a given signature.